src/test/com/Elessar/app/client/MyClientIntegTest.java

```diff
@@ -0,0 +1,233 @@
+package com.Elessar.app.client;
+
+import com.Elessar.app.server.Message;
+import com.Elessar.app.server.MyServer;
+import com.Elessar.app.server.User;
+import com.Elessar.app.util.MetricManager;
+import com.Elessar.database.MongoDB;
+import com.Elessar.database.MyDatabase;
+import com.Elessar.proto.Logoff.LogoffResponse;
+import com.Elessar.proto.Logon.LogonResponse;
+import com.Elessar.proto.P2Pmsg.P2PMsgResponse;
+import com.Elessar.proto.Registration.RegistrationResponse;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.IndexOptions;
+import com.mongodb.client.model.Indexes;
+import org.bson.Document;
+import org.junit.*;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Filters.or;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by Hans on 2/24/19.
+ */
+public class MyClientIntegTest {
+    private static MongoDatabase mongoDB;
+    private static MyClient clientA, clientB;
+    private static int clientA_Port, clientB_Port;
+
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        final String serverAddress = "127.0.0.1";
+        final int serverPort = 9000;
+        final String serverURL = new StringBuilder().append("http://")
+                                                    .append(serverAddress).append(":")
+                                                    .append(serverPort).toString();
+        clientA_Port = 4000;
+        clientB_Port = 5000;
+        final MetricManager serverMetricManager = new MetricManager("ServerMetric", 1000000);
+        final MetricManager clientMetricManager = new MetricManager("ClientMetric", 1000000);
+
+        // Setup server
+        mongoDB = MongoClients.create("mongodb://localhost:27017").getDatabase("test");
```

Comment:
better to use, say "MyClientIntegTest" as the database name so we won't accidentally mess up other tests' databases/collections